docs/webapp.adoc

```diff
@@ -1,30 +1,625 @@
+
+{zwsp} +
 
 === Source files
 
-.Files involved (src/main/resources/...):
+.Files involved (src/main/resources/...) - in addition to <<guillotine#chapter_source_files, the ones from chapter 8>>:
 [source,files]
 ----
+webapp/
+    webapp.es6
+    webapp.html
+
+assets/webapp/
+    script.es6
 ----
-<1> We're going to
+
+XP uses _webapp.es6_ and _webapp.html_ to generate an initial HTML that directly makes the browser run most of <<#nutshell, the steps above>>, fetching assets and setting up initial values, and then calling the final _script.es6_ asset, which handles the rest.
 
 {zwsp} +
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+{zwsp} +
+{zwsp} +
 {zwsp} +
+
+
+== 1. Completely standalone
+
+In this first of the two approaches, we'll minimize the use of runtime XP: all values and asset URLs are hardcoded. XP is used to serve the initial HTML and the assets, as well as providing a guillotine endpoint where the browser can fetch data, but this functionality can easily be replaced with any other file- and data-serving solution and still work just fine.
+
+[[html1]]
+=== HTML base
+
+The webapp begins with some basic HTML, setting it all up in the browser.
+
+.webapp.html:
+[source,html,options="nowrap"]
+----
+<!DOCTYPE html>
+<html>
+    <head>
+        <meta charset="UTF-8">
+        <title>Completely standalone</title>
+
+
+        <!--1-->
+        <script crossorigin src="https://unpkg.com/react@16/umd/react.production.min.js"></script>
+        <script crossorigin src="https://unpkg.com/react-dom@16/umd/react-dom.production.min.js"></script>
+
+        <!--2-->
+        <script src="(...) /react4xp/shared.d366a6c1b.js"></script>
+        <link rel="stylesheet" type="text/css" href="(...) /react4xp/shared.5440dda80.css" />
+        <link rel="stylesheet" type="text/css" href="(...) /react4xp/MovieList.css" />
+
+        <!--3-->
+        <script src="(...) /react4xp/MovieList.js"></script>
+
+        <style>
+            body { margin: 0; padding: 0; }
+            h1, p, .faux-spinner { padding: 30px; margin: 0 auto; font-family: 'DejaVu Sans', Arial, Helvetica, sans-serif; }
+        </style>
+    </head>
+
+    <body>
+        <h1>Top 3 movies to put in a list</h1>
+        <p>#4 will blow your mind!</p>
+
+        <!--4-->
+        <div id="movieListContainer">
+            <div class="faux-spinner">Loading movies...</div>
+        </div>
+
+        <!--5-->
+        <script>
+            var MOVIE_LIST_PARAMS= {
+                parentPath: '/moviesite',
+                apiUrl: '/admin/site/preview/default/draft/moviesite/api/headless',
```

Comment:
Don't we need to fetch this with serviceUrl?
This would not work on the master (published) branch
And I'm afraid people will copy paste this too.

Reply:
Usually, yes I agree. But in this case, an XP/thymeleaf-inserted view function like `serviceUrl` would defeat the whole purpose of the chapter: it's all bout doing without those, and getting stuff to work purely from a base HTML.

This path will often fail when copy-pasted, true. I tried to address that below point `<5>` below, but maybe I should put it under more emphasis or a separate bullet point?

Reply:
❓

Reply:
If you mentioned it I'm okay with it. Creating a service endpoint is still fully seperated in my opinion. 
Its creating the backend that sends over information. You still have to find/create an endpoint even if you connect your react app to any kind of database.